packages/back-end/src/services/stats.ts

```diff
@@ -32,153 +32,174 @@ import {
 import { QueryMap } from "../queryRunners/QueryRunner";
 import { MAX_ROWS_UNIT_AGGREGATE_QUERY } from "../integrations/SqlIntegration";
 
+// These same type definitions exist in gbstats.py
+export interface AnalysisSettingsForStatsEngine {
+  var_names: string[];
+  weights: number[];
+  baseline_index: number;
+  dimension: string;
+  stats_engine: string;
+  sequential_testing_enabled: boolean;
+  sequential_tuning_parameter: number;
+  difference_type: string;
+  phase_length_days: number;
+  alpha: number;
+  max_dimensions: number;
+}
+export interface MetricDataForStatsEngine {
+  metric: string;
+  rows: ExperimentMetricQueryResponseRows;
+  inverse: boolean;
+  multiple_exposures: number;
+}
+export interface DataForStatsEngine {
+  var_id_map: { [key: string]: number };
+  analyses: AnalysisSettingsForStatsEngine[];
+  metrics: MetricDataForStatsEngine[];
+}
```

Comment:
Refactored this data type to have two lists, which we will (effectively) double loop over in python to take the combination of each.

See related changes in `analyzeExperimentMetric` below.